src/pages/view/FilterBar/SaveFilterDialog.js

```diff
@@ -0,0 +1,95 @@
+import i18n from '@dhis2/d2-i18n'
+import {
+    Button,
+    Modal,
+    ModalContent,
+    ModalActions,
+    ButtonStrip,
+    ModalTitle,
+    InputField,
+    SingleSelectOption,
+    SingleSelectField,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React from 'react'
+import {
+    filterVisibility,
+    privateVisiblity,
+} from '../../../reducers/savedFilters.js'
+
+const SaveFilterDialog = ({
+    onCancel,
+    onConfirm,
+    filter = {},
+    showScope = true,
+}) => {
+    const [visibility, setVisibility] = React.useState(
+        filter.visibility || privateVisiblity
+    )
+    const [name, setName] = React.useState(filter.id ? filter.name : '')
+
+    const handleVisibilityChange = ({ selected }) => {
+        setVisibility(selected)
+    }
+    const handleNameChange = ({ value }) => {
+        setName(value)
+    }
+    const handleConfirm = () => {
+        onConfirm({ name, visibility, id: filter.id })
+    }
+
+    return (
+        <Modal
+            dataTest="save-filter-dialog"
+            onClose={onCancel}
+            small
+            position="middle"
+        >
+            <ModalTitle>{i18n.t('Save Filter')}</ModalTitle>
+            <ModalContent>
+                <InputField
+                    type="text"
+                    placeholder={i18n.t('Untitled filter')}
+                    label={i18n.t('Name')}
+                    value={name}
+                    onChange={handleNameChange}
+                />
+                {showScope && (
+                    <SingleSelectField
+                        label={i18n.t('Scope')}
+                        selected={visibility}
+                        onChange={handleVisibilityChange}
+                    >
+                        {filterVisibility.map((filter) => (
+                            <SingleSelectOption
+                                key={filter}
+                                value={filter}
+                                label={filter}
+                            />
+                        ))}
+                    </SingleSelectField>
+                )}
+            </ModalContent>
+            <ModalActions>
+                <ButtonStrip>
+                    <Button key="cancel" secondary onClick={onCancel}>
+                        {i18n.t('Cancel')}
+                    </Button>
+
+                    <Button key="confirm" primary onClick={handleConfirm}>
```

Comment:
I think we need the loading indicator here as well, right? For slow connections and to prevent user clicking twice.